### ISSUE TITLE
PostgreSQL commands not working (directory permissions)

### ISSUE THREAD

**Issue by tschni** [drud/ddev-contrib#183]
Hi all, 

I noticed that the PostgreSQL commands (https://github.com/drud/ddev-contrib/tree/master/docker-compose-services/postgres/commands/postgres) do not work out of the box.

I was able to find the reason relatively quickly: In the comments the export & import directory ".ddev/pgsql-db" is mentioned, this is also set correctly during pg_dump and import, but the pgsql_export script still says that the directory "/mnt/ddev_config/import-db" and not "pgsql-db" should be created.

When I changed this to "pgsql-db" everything worked as well. However, the "pgsql-db" folder is not included in .gitignore, so I wonder if it doesn't make sense again to store the database dump in "import-db".

**Comment by rfay** [comment]
@felmab, @tyler36 , @penance316 @penyaskito @pcambra @dacostafilipe

**Comment by penyaskito** [comment]
@tschni I never used `.ddev/pgsql-db`myself. Is this for auto-importing the db when running `ddev start`? 
Can you provide concrete steps to reproduce? Which commands are you executing?

I'm in the process of switching from `image: mdillon/postgis:11` to `image: postgres:12` on one of my projects and I found some issues myself, but still not clear if related. In my case I had to apply this patch:

```
diff --git a/.ddev/commands/postgres/pgsql_restore b/.ddev/commands/postgres/pgsql_restore
index c94390f..304d881 100755
--- a/.ddev/commands/postgres/pgsql_restore
+++ b/.ddev/commands/postgres/pgsql_restore
@@ -8,4 +8,4 @@
 su postgres -c "psql -U db -c 'DROP SCHEMA public CASCADE; CREATE SCHEMA public; GRANT ALL ON SCHEMA public TO db; GRANT ALL ON SCHEMA public TO public;'"

 # Import via user postgres to avoid credentials prompt
-su postgres -c "pg_restore --no-owner -U db -d db /mnt/ddev_config/import-db/postgresql.db.dump"
+su postgres -c "pg_restore --host=localhost --no-owner -U db -d db /mnt/ddev_config/import-db/postgresql.db.dump"
```

I might open a different issue for this, but wondering out if doing a similar thing in the command you are executing changes anything at all?

**Comment by felmab** [comment]
@tschni apparently, yes, I also had to create a folder manually for the export command to work, sorry for not reporting it. Since you already fixed it locally, maybe you could send a PR? Keeping import-db instead of pgsql-db would make sense IMHO.

**Comment by tschni** [comment]
> @tschni apparently, yes, I also had to create a folder manually for the export command to work, sorry for not reporting it. Since you already fixed it locally, maybe you could send a PR? Keeping import-db instead of pgsql-db would make sense IMHO.

No issue, I'll create PR in a moment :)

**Comment by tschni** [comment]
> @tschni I never used `.ddev/pgsql-db`myself. Is this for auto-importing the db when running `ddev start`? Can you provide concrete steps to reproduce? Which commands are you executing?
> 
> I'm in the process of switching from `image: mdillon/postgis:11` to `image: postgres:12` on one of my projects and I found some issues myself, but still not clear if related. In my case I had to apply this patch:
> 
> ```
> diff --git a/.ddev/commands/postgres/pgsql_restore b/.ddev/commands/postgres/pgsql_restore
> index c94390f..304d881 100755
> --- a/.ddev/commands/postgres/pgsql_restore
> +++ b/.ddev/commands/postgres/pgsql_restore
> @@ -8,4 +8,4 @@
>  su postgres -c "psql -U db -c 'DROP SCHEMA public CASCADE; CREATE SCHEMA public; GRANT ALL ON SCHEMA public TO db; GRANT ALL ON SCHEMA public TO public;'"
> 
>  # Import via user postgres to avoid credentials prompt
> -su postgres -c "pg_restore --no-owner -U db -d db /mnt/ddev_config/import-db/postgresql.db.dump"
> +su postgres -c "pg_restore --host=localhost --no-owner -U db -d db /mnt/ddev_config/import-db/postgresql.db.dump"
> ```
> 
> I might open a different issue for this, but wondering out if doing a similar thing in the command you are executing changes anything at all?

Hey, yes the directory is for the auto-import feature.

However, the directory was not created with appropriate permissions before, should be fixed by my PR #184 now.

I don't use the pgsql_restore at all, but it looks like a different issue to me.
